test/main.test.js

```diff
@@ -129,4 +130,50 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('doesn\'t mix up contexts if using concurrently then() from when 3.7.7',
+  function() {
+    expect(require('when/package.json').version).to.equal('3.7.7');
+    var timeout = 50;
+    // Concurrent execution number 1 of 2
+    var execution1 = new Promise(function execution1(outerResolve, reject) {
+      LoopBackContext.runInContext(function pushToContext1() {
```

Comment:
Problem solved without changing the tests, since the change is no longer in the middleware [factory] (I moved it inside the original method, instead of patching it in the middleware [factory])